evolvedbinary/hsg-shell
# Feature/image viewer 226

### Create a new page image view of a scanned publication
Design a new page image view of a scanned publication, which displays a link to the corresponding document(s) that appear on the page.

- [x] Install iiif cantaloupe as local image server for local development
- [x] Configure cantaloupe.properties
- [x] Implement Openseadragon as a viewer
- [x] Configure Openseadragon
- [x] Customize OSD
- [x] Integrate Openseadragon in build automation
- [x] Update image paths for scans in `pages.xql, frus-ajax.xql`
- [x] Update layout / css changes
- [x] Fix Ajax navigation (side-page-navigation)
- [x] Create CI tests
- [x] Create a back-link to the corresponding volume page(s)

---

### Testlinks
#### Local:
* http://localhost:8080/exist/apps/hsg-shell/historicaldocuments/frus1902app1/pg_11 
* http://localhost:8080/exist/apps/hsg-shell/historicaldocuments/frus1969-76v18/pg_71

#### Testserver:
* http://kiwi.evolvedbinary.com:8080/exist/apps/hsg-shell/historicaldocuments/frus1902app1/pg_11
* http://kiwi.evolvedbinary.com:8080/exist/apps/hsg-shell/historicaldocuments/frus1969-76v18/pg_71

#### Corresponding links in production with current static PNG files on history.state.gov:
* https://history.state.gov/historicaldocuments/frus1902app1/pg_11 
* https://history.state.gov/historicaldocuments/frus1969-76v18/pg_71

### Test setup on local machine
You need to install and run a local Cantaloupe image server in order to process and serve IIIF images while testing the application on your local machine.

1. Download the current release from https://github.com/cantaloupe-project/cantaloupe/releases, and unzip
2. Copy the readily modified `cantaloupe.properties` file from this PR https://github.com/evolvedbinary/hsg-shell/blob/4af6e073aa25afbd1ffca7189a43e33a946e88e4/cantaloupe.properties and paste it to your Cantaloupe installation. See also https://cantaloupe-project.github.io/manual/5.0/getting-started.html
3. Start Cantaloupe as described in the manual (note that it requires Java 11) - by running `java -Dcantaloupe.config=cantaloupe.properties -Xmx2g -jar cantaloupe-5.0.5.jar` 
4. The cantaloupe port should be listening to 8182 now, which you can check by requesting e.g. http://localhost:8182/iiif/3/frus1969-76v18%2Ftiff%2F0101.tif/info.json for checking the json, or the admin GUI at  http://localhost:8182/admin

---


New Layout with border around viewer and background-color when zooming out at maximum:
<img width="637" alt="Screenshot 2022-10-19 at 16 47 32" src="https://user-images.githubusercontent.com/1157669/196724999-6c2c84c0-0d3d-442c-b35e-6b677c4050c6.png">

---

<img width="644" alt="Screenshot 2022-10-17 at 18 44 22" src="https://user-images.githubusercontent.com/1157669/196235639-7426adcc-c74b-4ced-a3e6-d7bb14688818.png">
